### ISSUE TITLE
Fix systematic insurance mispricing in pricing simulation notebooks

### ISSUE THREAD

**Issue by AlexFiliakov** [AlexFiliakov/Ergodic-Insurance-Limits#203]
## Description

Critical bug in insurance loss calculation formula causing systematic mispricing of insurance premiums. The current formula incorrectly applies the deductible AFTER capping at policy limit, leading to understatement of insured losses by the full deductible amount ($100K) for all losses exceeding the policy limit.

## Context

**Project Area**: Loss modeling and insurance pricing simulations  
**Complexity**: 2/10 (simple formula correction across multiple files)  
**Impact**: High - distorts optimal insurance limit selection, making lower limits appear artificially attractive

### Current (INCORRECT) Formula
```python
insured_loss = max(min(loss_event.amount, policy_limit) - deductible, 0)
```

### Correct Formula
```python
insured_loss = max(min(loss_event.amount - deductible, policy_limit), 0)
```

### Impact Example

With $50M policy limit and $100K deductible:

| Loss Amount | Current Formula | Correct Formula | Understatement |
|-------------|----------------|-----------------|----------------|
| $60M | $49.9M | $50M | **$100K** |
| $100M | $49.9M | $50M | **$100K** |
| $500M | $49.9M | $50M | **$100K** |

### Why This Matters

With heavy-tailed loss distributions (Pareto α=2.5):
- **Lower policy limits** have MORE losses exceeding them → many mispriced losses → systematic underpricing
- **Higher policy limits** have fewer losses exceeding them → less underpricing
- **Result**: $50M limit appears optimal when $200M should be optimal

### Affected Files (4 files, ~6 instances)

1. `ergodic_insurance/notebooks/results_tail_sim_01/run_tail_simulation_colab.py:203`
   - Pricing simulation for tail risk analysis
   
2. `ergodic_insurance/notebooks/run_basic_simulation.py:124`
   - Basic simulation notebook runner
   
3. `ergodic_insurance/notebooks/ergodicity_basic_simulation.ipynb` (line ~71015)
   - Main exploratory notebook
   
4. `ergodic_insurance/papers/2025-09-research-paper-1/ergodicity_basic_simulation.ipynb` (lines ~181, ~561)
   - Research paper notebook (2 instances)

### Note on Core Library

The `InsuranceLayer.calculate_recovery()` method in `ergodic_insurance/insurance.py:130-166` is **correct** and properly implements the formula. Only the pricing simulation notebooks contain the bug.

## Requirements

1. Fix the insured loss calculation formula in all affected files
2. Verify the fix produces correct results by running test simulations
3. Ensure consistency across all notebooks and scripts
4. Update any related documentation if needed

## Implementation Steps

- [ ] Update formula in `ergodic_insurance/notebooks/results_tail_sim_01/run_tail_simulation_colab.py:203`
- [ ] Update formula in `ergodic_insurance/notebooks/run_basic_simulation.py:124`
- [ ] Update formula in `ergodic_insurance/notebooks/ergodicity_basic_simulation.ipynb` (search for the pattern)
- [ ] Update formula in `ergodic_insurance/papers/2025-09-research-paper-1/ergodicity_basic_simulation.ipynb` (2 instances)
- [ ] Run pricing simulation with corrected formula to verify results
- [ ] Compare new efficiency frontier with the buggy version to confirm expected behavior
- [ ] Document the change in commit message with clear before/after explanation

## Acceptance Criteria

- ✅ All instances of the incorrect formula are replaced with the correct formula
- ✅ Pricing simulations run successfully with corrected formula
- ✅ Efficiency frontier shows higher policy limits as more optimal (e.g., $200M vs $50M)
- ✅ Insured losses for losses exceeding policy limit are correctly calculated (no $100K understatement)
- ✅ Code passes all existing tests
- ✅ Changes are committed with descriptive message explaining the fix

## Additional Context

This bug was discovered while investigating why the efficiency frontier analysis showed $50M as the optimal policy limit when theory and ergodic analysis suggested $200M should be optimal. The systematic underpricing of lower limits created an artificial pricing subsidy that distorted the optimization results.
